core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -504,16 +510,23 @@ private void registerBuiltinSorts() {
         namedWriteableRegistry.register(SortBuilder.class, FieldSortBuilder.NAME, FieldSortBuilder::new);
     }
 
-    private void registerBuiltinFunctionScoreParsers() {
-        registerFunctionScoreParser(new ScriptScoreFunctionParser());
-        registerFunctionScoreParser(new GaussDecayFunctionParser());
-        registerFunctionScoreParser(new LinearDecayFunctionParser());
-        registerFunctionScoreParser(new ExponentialDecayFunctionParser());
-        registerFunctionScoreParser(new RandomScoreFunctionParser());
-        registerFunctionScoreParser(new FieldValueFactorFunctionParser());
+    private void registerBuiltinScoreFunctionParsers() {
+        registerScoreFunction(ScriptScoreFunctionBuilder.FIELD, ScriptScoreFunctionBuilder::fromXContent, ScriptScoreFunctionBuilder.NAME,
```

Comment:
FUNCTION_NAME_FIELD should be fine. I can switch.

I like when function parameters are last in order of arguments because it makes it look nicer when you define the function right there. But in this case we're not really going to do that. There are two functional references in the arguments anyway. So I'll go with your order.

What do you think about passing both name and FUNCTION_NAME_FIELD? I see you settled on not doing that for queries and instead documenting that the primary in the QUERY_NAME_FIELD was always registered. Should I do that?